doc/maintenance.md

```diff
@@ -0,0 +1,539 @@
+
+#### Introduction
+
+The usage of the RegistryInterface messages is foreseen via the POST verb, under resources:
+- ```/rest/subscription``` for Subscriptions
+- ```/rest/registration``` for Registrations
+- ```/rest``` for any RegistryInterface message
+
+The available ```RegistryInterface``` messages are the following:
+- ```SubmitRegistrationsRequest```
+- ```SubmitRegistrationsResponse```
+- ```SubmitSubscriptionsRequest```
+- ```SubmitRegistrationsResponse```
+- ```QueryRegistrationRequest```
+- ```QueryRegistrationResponse```
+- ```QuerySubscriptionRequest```
+- ```QuerySubscriptionResponse```
+
+In addition, the above shall remain as part of the generic ```RegistryInterface``` message.
+
+
+#### Retrieving Subscriptions and Registrations
+
+For Subscriptions:
+- The client may POST a ```QuerySubscriptionRequest``` or a ```RegistryInterface/QuerySubscriptionRequest``` message under resource ```/rest/subscription``` for retrieving Subscriptions. 
+- As a result, a ```QuerySubscriptionResponse``` or a ```RegistryInterface/QuerySubscriptionResponse``` message will be returned as a response to describe the requested Subscriptions.
+
+Similarly, for Registrations:
+- The client may POST a ```QueryRegistrationRequest``` or a ```RegistryInterface/QueryRegistrationRequest``` message under resource ```/rest/registration``` for retrieving Registrations.
+- As a result, a ```QueryRegistrationResponse``` or a ```RegistryInterface/QueryRegistrationResponse``` message will be returned as a response to describe the requested Registrations.
+
+#### Maintaining Subscriptions and Registrations
+
+For Subscriptions:
+- The client may POST a ```SubmitSubscriptionsRequest``` or a ```RegistryInterface/SubmitSubscriptionsRequest``` message under resource ```/rest/subscription``` for inserting/updating/deleting Subscriptions. 
+- As a result, a ```SubmitSubscriptionsResponse``` or a ```RegistryInterface/SubmitSubscriptionsResponse``` message will be returned as a response to describe the outcome of the insertion/update/deletion of Subscriptions.
+
+Similarly, for Registrations:
+- The client may POST a ```SubmitRegistrationsRequest``` or a ```RegistryInterface/SubmitRegistrationsRequest``` message under resource ```/rest/registration``` for inserting/updating/deleting Registrations. 
+- As a result, a ```SubmitRegistrationsResponse``` or a ```RegistryInterface/SubmitRegistrationsResponse``` message will be returned as a response to describe the outcome of the insertion/update/deletion of Registrations.
+
+
+### ANNEX II - Structure response message
+
+The `SubmitStructureResponse` message must be returned in any case (success, partial success, failure).
+This is defined as part of the `RegistryInterface` messages.
+This message includes the following information per submitted Artefact:
+- The `action`, e.g. `Append`, `Replace` or `Delete` (`Information` is also available)
```

Comment:
Is this about the result of the action performed by the server per artefact? In which case would 'Information' to be used?